src/main/kotlin/it/pagopa/ecommerce/payment/requests/errorhandling/ExceptionHandler.kt

```diff
@@ -134,45 +134,79 @@ class ExceptionHandler(@Value("#{\${fields_to_obscure}}") val fieldToObscure: Se
-          HttpStatus.BAD_GATEWAY)
-      } else if (Arrays.stream(ValidationFaultDto.values()).anyMatch { z ->
+          HttpStatus.SERVICE_UNAVAILABLE)
+      } else if (Arrays.stream(ValidationFaultPaymentUnavailableDto.values()).anyMatch { z ->
         z.value == faultCode
       }) {
         ResponseEntity(
-          ValidationFaultPaymentProblemJsonDto(
+          ValidationFaultPaymentUnavailableProblemJsonDto(
             title = "Validation Fault",
-            faultCodeCategory = FaultCategoryDto.PAYMENT_UNKNOWN,
-            faultCodeDetail = ValidationFaultDto.valueOf(faultCode)),
-          HttpStatus.NOT_FOUND)
-      } else if (Arrays.stream(GatewayFaultDto.values()).anyMatch { z -> z.value == faultCode }) {
-        ResponseEntity(
-          GatewayFaultPaymentProblemJsonDto(
-            title = "Payment unavailable",
-            faultCodeCategory = FaultCategoryDto.GENERIC_ERROR,
-            faultCodeDetail = GatewayFaultDto.valueOf(faultCode)),
+            faultCodeCategory =
+              ValidationFaultPaymentUnavailableProblemJsonDto.FaultCodeCategory.PAYMENT_UNAVAILABLE,
+            faultCodeDetail = ValidationFaultPaymentUnavailableDto.valueOf(faultCode)),
           HttpStatus.BAD_GATEWAY)
-      } else if (Arrays.stream(PartyTimeoutFaultDto.values()).anyMatch { z ->
+      } else if (Arrays.stream(PaymentOngoingStatusFaultDto.values()).anyMatch { z ->
         z.value == faultCode
       }) {
         ResponseEntity(
-          PartyTimeoutFaultPaymentProblemJsonDto(
-            title = "Gateway Timeout",
-            faultCodeCategory = FaultCategoryDto.GENERIC_ERROR,
-            faultCodeDetail = PartyTimeoutFaultDto.valueOf(faultCode)),
-          HttpStatus.GATEWAY_TIMEOUT)
-      } else if (Arrays.stream(PaymentStatusFaultDto.values()).anyMatch { z ->
+          PaymentOngoingStatusFaultPaymentProblemJsonDto(
+            title = "Payment Status Fault",
+            faultCodeCategory =
+              PaymentOngoingStatusFaultPaymentProblemJsonDto.FaultCodeCategory.PAYMENT_ONGOING,
+            faultCodeDetail = PaymentOngoingStatusFaultDto.valueOf(faultCode)),
+          HttpStatus.CONFLICT)
+      } else if (Arrays.stream(PaymentExpiredStatusFaultDto.values()).anyMatch { z ->
         z.value == faultCode
       }) {
         ResponseEntity(
-          PaymentStatusFaultPaymentProblemJsonDto(
+          PaymentExpiredStatusFaultPaymentProblemJsonDto(
             title = "Payment Status Fault",
-            faultCodeCategory = FaultCategoryDto.PAYMENT_UNAVAILABLE,
-            faultCodeDetail = PaymentStatusFaultDto.valueOf(faultCode)),
+            faultCodeCategory =
+              PaymentExpiredStatusFaultPaymentProblemJsonDto.FaultCodeCategory.PAYMENT_EXPIRED,
+            faultCodeDetail = PaymentExpiredStatusFaultDto.valueOf(faultCode)),
           HttpStatus.CONFLICT)
+      } else if (Arrays.stream(PaymentCanceledStatusFaultDto.values()).anyMatch { z ->
```

Comment:
instead of this if else chain, what about use [this approch](https://github.com/pagopa/pagopa-ecommerce-payment-requests-service/pull/96/files#diff-61b63991dfc95f51d0491a4ee6602d4efe2eac5ee30f83bd5fc7f0c865d6203fR129-R197)?
This was from an old pr that I opened in the past to fix a misalignment, the key point here is to start from the fault code enumerations and associate to each the expected fault code category, http status code and so on.
I think it can be a little easier to read and can let discover duplicated fault code mappings easily (performing tests against all fault codes as done [here](https://github.com/pagopa/pagopa-ecommerce-payment-requests-service/pull/96/files#diff-c815c66ada0267de914e6deddfca6da3b905df0cc6de78866b254a39efe322f8R133-R134)  for example)

Reply:
whit this approach I found in as I opened the referenced PR that GENERIC_ERROR was associated to both 502 and 504 error responses

Reply:
done, thanks!